text/0232-maturity-of-otel.md

```diff
@@ -0,0 +1,151 @@
+
+The Collector's [stability levels](https://github.com/open-telemetry/opentelemetry-collector#stability-levels) inspired the maturity levels.
+
+## Motivation
+
+Quite often, the community is faced with the question of the quality and maturity expectations of its diverse set of components. This OTEP aims to bring clarity by establishing a framework to communicate the project's maturity. As the OpenTelemetry project comprises a multitude of SIGs, and each SIG has several components of varying quality, having this framework will help set the right expectations for OpenTelemetry users.
+
+Only once we reach the "stable" level at the project level will we be ready to consider graduating OpenTelemetry at the CNCF, with all of the Tier-1 SIGs being in the context of the graduation, including their Tier-1 Components. This OTEP provides a framework for describing project maturity as a whole.
+
+## Glossary
+
+* *SIG:* an official SIG within the OpenTelemetry project, like the Collector or Java. Some WGs are temporarily accepted, given they should be SIGs instead of WGs, like the Semantic Conventions WG.
+* *Tier-1 SIG:* a SIG within the OpenTelemetry project deemed critical by the GC/TC.
+* *Candidate Tier-1 SIG:* a SIG that is deemed as critical for the future of the project, but might not be of the same maturity level yet. It's expected to reach that level eventually, and other Tier-1 SIGs should be prepared to accommodate the new component as needed.
+* *Component:* module maintained by a SIG like a Collector exporter or an SDK sampler.
+* *Tier-1 Component:* a component selected by project maintainers as critical for the SIG. For instance, the OTLP receiver will likely be selected as a Tier-1 Component of the Collector. See Tier-1 Components for more details.
+* *Candidate Tier-1 Component:* similar to "Candidate Tier-1 SIG", but for a component within a SIG.
+* *Dependency:* an OpenTelemetry SIG or component that is required by another SIG or component. For instance, the OTLP Logs Data Model from the Spec is required for the Collector's Logging support.  
+
+## Explanation
+
+### Maturity levels
+
+SIGs and components of a project MUST have a declared maturity level established by SIG maintainers (SIGs). The maturity level for a SIG or component is, at most, the lowest level of its dependencies or priority components. For instance, if the Collector SIG maintainers declare the "otlpreceiver" component as a tier-1 component of the Collector core distribution, and the "otlpreceiver" declares a dependency on the OpenTelemetry Collector API "config" package which is marked with a maturity level of "beta", the "otlpreceiver" can be at most "beta". The OpenTelemetry Collector "core" distribution is also affected so that it can be at most "beta".
```

Comment:
> The OpenTelemetry Collector "core" distribution is also affected so that it can be at most "beta".

I disagree with this. In java, we bundle many components of varying stability with the java agent. Waiting for all those components to be stable for the agent to be stable would mean that the agent is never stable. Instead, we have strategies like disabling certain less mature instrumentation by default.

Reply:
Good point. As a user of the Java agent, am I right to have the expectation that things will not break between versions when it comes to their configuration and telemetry. If I'm expected to deal with changed in metric names when migrating between versions, can I opt-out from those changes or should I just assume those metrics are not stable for me to consume it in alerts/dashboards?

Reply:
I thin individual instrumentation modules will each have their own stability designations, and so user expectations will have to be on a case by case basis. I think the @open-telemetry/java-instrumentation-maintainers will have more thoughts about how to about stability guarantees of an agent comprised of many instrumentation modules.